src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -185,18 +200,18 @@ private void LoadItems(PackageItemListViewModel selectedPackageItem)
 
             token.ThrowIfCancellationRequested();
 
-            // multiple loads may occur at the same time
-            if (currentLoader == _loader)
-            {
-                UpdatePackageList(loadedItems.ToList(), refresh: false);
-            }
-
-            token.ThrowIfCancellationRequested();
-
             await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
                 _loadingStatusBar.ItemsLoaded = currentLoader.State.ItemsCount;
+                
+                // multiple loads may occur at the same time
+                if (currentLoader == _loader)
```

Comment:
I think this `if` should be around `RunAsync` at line 203. It doesn't make sense updating UI in general if loader is deprecated.